common/src/main/java/com/wynntils/features/inventory/CustomBankPageNamesFeature.java

```diff
@@ -0,0 +1,112 @@
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.INVENTORY)
+public class CustomBankPageNamesFeature extends Feature {
+    private TextInputBoxWidget editInput;
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent event) {
+        if (Models.Bank.getCurrentContainer() == null) return;
+        AbstractContainerScreen<?> screen = (AbstractContainerScreen<?>) event.getScreen();
+
+        // This is screen.topPos and screen.leftPos, but they are not calculated yet when this is called
+        int renderX = (screen.width - screen.imageWidth) / 2;
+        int renderY = (screen.height - screen.imageHeight) / 2;
+
+        screen.addRenderableWidget(screen.addRenderableWidget(new ContainerEditNameButton(
+                renderX + (screen.imageWidth - screen.titleLabelX) - 10, renderY + (screen.titleLabelY) - 4, 6, 16)));
+    }
+
+    @SubscribeEvent
+    public void onRenderLabels(ContainerLabelRenderEvent event) {
+        if (Models.Bank.getCurrentContainer() == null) return;
+
+        if (Models.Bank.isEditingName()) {
+            event.setCanceled(true);
+            renderEditInput(event.getScreen());
+        } else {
+            if (editInput != null) {
+                event.getScreen().removeWidget(editInput);
+                editInput = null;
+            }
+
+            if (!Models.Bank.getPageName(Models.Bank.getCurrentPage()).isEmpty()) {
+                event.setCanceled(true);
+                renderPageName(event.getPoseStack(), event.getScreen());
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public void onInventoryKeyPress(InventoryKeyPressEvent event) {
+        if (event.getKeyCode() != GLFW.GLFW_KEY_ENTER) return;
+        if (!Models.Bank.isEditingName()) return;
+
+        Models.Bank.saveCurrentPageName(editInput.getTextBoxInput());
+    }
+
+    @SubscribeEvent
+    public void onSlotClicked(ContainerClickEvent e) {
+        if (Models.Bank.getCurrentContainer() == null) return;
+        Models.Bank.toggleEditingName(false);
+    }
+
+    private void renderEditInput(AbstractContainerScreen<?> screen) {
+        if (editInput != null) return;
+
+        editInput = new TextInputBoxWidget(
+                screen.leftPos + screen.titleLabelX,
+                screen.topPos + screen.titleLabelY,
+                screen.imageWidth - 50,
+                FontRenderer.getInstance().getFont().lineHeight,
+                null,
+                (ScreenExtension) screen);
+
+        editInput.setTextBoxInput(Models.Bank.getPageName(Models.Bank.getCurrentPage()));
+
+        screen.addRenderableWidget(editInput);
+    }
+
+    private void renderPageName(PoseStack poseStack, AbstractContainerScreen<?> screen) {
```

Comment:
You didn't change this? You can use the event.

Reply:
Doing that now, you missed passing the instance into the original call so it wasn't working ;)

Reply:
Oh, right. It doesn't complain until you run it.. I only compiled it.